Issue 5048 - Support for nsslapd-tcp-fin-timeout and nsslapd-tcp-keepalive-time

miztake opened the pull request:
Description:
Installing 389-ds modifies system parameters by 70-dirsrv.conf.
"net.ipv4.tcp_fin_timeout" and "net.ipv4.tcp_keepalive_time" can be set for
389-ds sockets using setsockopt(). System parameters should not be changed
as much as possible and should only be applied to 389-ds sockets.

Fix Description:
To set parameters for 389-ds sockets, following two attributes have been added.
- nsslapd-tcp-fin-timeout
- nsslapd-tcp-keepalive-time

"net.ipv4.tcp_fin_timeout" and "net.ipv4.tcp_keepalive_time" of 70-dirsrv.conf
are no longer needed.

Relates: https://github.com/389ds/389-ds-base/issues/5048

Firstyear commented on ldap/servers/slapd/daemon.c:
```diff
@@ -2561,6 +2563,24 @@ configure_pr_socket(PRFileDesc **pr_socket, int secure, int local)
     } /* else (!enable_nagle) */
 
+    if (!local) {
+#ifdef TCP_LINGER2
```
I don't think we need the TCP_ defs here.

miztake replied:
Dear @Firstyear,

Thank you for your comment.
I have removed the TCP_ defs .

miztake committed:
Issue 5048 - Support for nsslapd-tcp-fin-timeout and nsslapd-tcp-keepalive-time

Description:
Installing 389-ds modifies system parameters by 70-dirsrv.conf.
"net.ipv4.tcp_fin_timeout" and "net.ipv4.tcp_keepalive_time" can be set for
389-ds sockets using setsockopt(). System parameters should not be changed
as much as possible and should only be applied to 389-ds sockets.

Fix Description:
To set parameters for 389-ds sockets, following two attributes have been added.
- nsslapd-tcp-fin-timeout
- nsslapd-tcp-keepalive-time
"net.ipv4.tcp_fin_timeout" and "net.ipv4.tcp_keepalive_time" of 70-dirsrv.conf
are no longer needed.

Relates: https://github.com/389ds/389-ds-base/issues/5048

Reviewed by: Firstyear, mreynolds389 (Thanks!)

Firstyear reviewed (approved):
Looks great, thank you! @mreynolds389 want to check too? This is to move some of our sysctl tunings into the server so we don't affect the running system.
